app/users/user-queries.js

```diff
@@ -10,10 +11,65 @@ const getAllUsers = () => User.find({})
     .lean()
     .exec();
 
-const updateUserById = (id, update) => User.findByIdAndUpdate(id, update, {
-    new: true,
+const getPushValues = (eventsAttended, eventsFavorited, charitiesFavorited) => ({
+    eventsAttended:
+      eventsAttended.length > 0
+          ? {
+              $each: [...eventsAttended],
+          }
+          : null,
+    eventsFavorited:
+      eventsFavorited.length > 0
+          ? {
+              $each: [...eventsFavorited],
+          }
+          : null,
+    charitiesFavorited:
+      charitiesFavorited.length > 0
+          ? {
+              $each: [...charitiesFavorited],
+          }
+          : null,
 });
 
+const updateUserById = async (id, update) => {
+    let updatedUser = {};
+
+    if (
+        _.get(update, 'eventsFavorited', []).length > 0
+    || _.get(update, 'eventsAttended', []).length > 0
+    || _.get(update, 'charitiesFavorited', []).length > 0
+    ) {
+        const updateObj = { ...update };
```

Comment:
This should be a let as well since we are mutating. Can we change the name of this  to something like `updateCopy`

Reply:
done